java/src/jmri/jmrit/ToolsMenu.java

```diff
@@ -172,6 +191,152 @@ public ToolsMenu() {
         add(new JSeparator());
         // LogixNG menu
         add(new jmri.jmrit.logixng.tools.swing.LogixNGMenu());
+
+        updateProgrammerStatus(null);
```

Comment:
I think it would be good to have a comment like
`// Enable or disable the service mode programmer menu items for the types of programmer available.`
before this section of `addPropertyChangeListener`s so that it's clear what they are doing.